slackbridge/messages.py

```diff
@@ -55,7 +56,32 @@ def resolve(self):
         if not channel_id or not isinstance(channel_id, str):
             return
 
-        if channel_id in self.bridge_bot.channels:
+        if channel_id[0] == 'D':  # DM channels start with a D
+            if not user_bot.im_id:
+                user_bot.im_id = channel_id
+
+            # get irc target name from message
+            if 'text' in self.raw_message:
+                match = re.search('(.*\w):', self.raw_message['text'])
```

Comment:
I'm concerned that this will match more than just a IRC nickname and PM to some non-existent bot and the message will get lost. In your testing, was that a problem at all?

Also, do you know of a way to check if the IRC user is actually online and respond with something if they aren't available so that people on Slack know if their message hasn't been received due to the user leaving IRC or something? That seems like one of the main usability concerns here.

Reply:
I haven't tested enough to know if this regex will overmatch.

This is related to another problem where we need to verify if the user is authenticated via NickServ. 

A possible solution I thought of is calling /whois [user], saving the post_to_irc call as a callback which is maintained in a dict keyed by the IRC username, then executing it only if the whois reply indicates that the user is online and authenticated otherwise respond to the Slack user as necessary.

Do you see any immediate problems with this?

Reply:
Hmm, no that should work, I imagine a whois query per message would not be too bad in terms of load or delay, but that's something that more testing could help with. I think this would be a useful feature for public messages too, like where an IRC user is directly mentioned but isn't online in a public channel.

I don't think the authentication is a big deal, because while users can impersonate each other, some frequent users of IRC still aren't authenticated, although I do see that it would matter more with private messages that are potentially confidential.

Reply:
This is odd regex... I assume this is to get an IRC nick? A bit to lax according to the RFC.

We have the [strip_nick method in utils](https://github.com/ocf/slackbridge/blob/5e1945b9a09fa8189cfdb420f058be58d6849eb0/slackbridge/utils.py#L69)